### PR TITLE
Making a clear distinction between solrapi-client-api and -spi

### PR DESCRIPTION
**Pull request by tgeens**
This is a first of a series of PRs.

* the module `alfresco-solrapi-client-spi` is renamed into `alfresco-solrapi-client-api`
* packages have been restructured

No functional changes

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -29,7 +29,7 @@ dockerCompose {
     dockerComposeWorkingDirectory = './docker-compose'
 
     // For running integration test during local development
-    // stopContainers = false
```
Comment (kerkhofsd) [inline build.gradle]: :arrow_up:

File: alfresco-remote-api-exceptions/build.gradle [hunk 2]
```diff
@@ -5,6 +5,10 @@ plugins {
 group = rootProject.group
 version = rootProject.version
 
+repositories {
```
Comment (kerkhofsd) [inline alfresco-remote-api-exceptions/build.gradle:8]: not necessary?

**Review by kerkhofsd — CHANGES_REQUESTED** [review]

File: alfresco-remote-api-exceptions/build.gradle (hunk 2 above)
Comment (tgeens) [reply alfresco-remote-api-exceptions/build.gradle:8]: I get warnings trying to resolve lombok if I don't have this.

File: build.gradle (hunk 1 above)
Comment (tgeens) [reply build.gradle]: Fixed

**Review by kerkhofsd — APPROVED** [review]

**Pull request merged**